src/__tests__/lib/rules/prefer-in-document.js

```diff
@@ -0,0 +1,67 @@
+/* eslint-disable no-template-curly-in-string */
+/**
+ * @fileoverview Prefer toBeInTheDocument over querying and asserting length.
+ * @author Anton Niklasson
+ */
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+import { RuleTester } from "eslint";
+import { queries } from "@testing-library/dom";
+import * as rule from "../../../rules/prefer-in-document";
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({ parserOptions: { ecmaVersion: 2015 } });
+ruleTester.run("prefer-in-document", rule, {
+  valid: [
+    ...Object.keys(queries)
+      .map((q) => [
+        `expect(screen.${q}('foo')).toBeInTheDocument()`,
+        `expect(${q}('foo')).toBeInTheDocument()`,
+        `expect(wrapper.${q}('foo')).toBeInTheDocument()`,
+      ])
+      .flat(),
+    {
+      code: `expect(screen.notAQuery('foo-bar')).toHaveLength(1)`,
+      errors: [
+        {
+          messageId: "useDocument",
+        },
+      ],
+    },
+    {
+      code: `expect(screen.getByText('foo-bar')).toHaveLength(2)`,
+      errors: [
+        {
+          messageId: "useDocument",
+        },
+      ],
+    },
+  ],
+  invalid: [
+    ...Object.keys(queries)
```

Comment:
actually thinking about this there's a couple changes I think we should make here.

1) split out the get, query and find queries.
2) find queries should use await
3) "query" queries should be checking that `.toHaveLength(0)` gets reported and fixed to `.not.toBeInTheDocument()`

Reply:
Yeah, I was just looking into the `await` case. Ah, right. Since the other ones throw instead? I'll look into it :)

Reply:
@AntonNiklasson I thought of another couple of use cases: `expect(screen.queryByText()).toBeNull()` should be changed to `.toBeInTheDocument()` and `.not.toBeNull()` should be changed to `.toBeInTheDocument()` and conversely we should check `.toBeDefined()` as well.

Reply:
Yeah, good catch! I've just pushed a few changes to split the queries. I've also added some failing test cases for `queryBy*` and `queryAllBy*`. I'll get back to passing the tests in a day or two.

Reply:
I've implemented some more logic to support more matchers than just `toHaveLength` now. Also the negation flipping from `.not.toBeNull` to `toBeInTheDocument`. Perhaps there's more matcher I haven't thought of?